### PR TITLE
Add Client.fetch_application

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

This PR adds a `fetch_application()` method to `Client` and a matching `get_application()` in http.py. This method allows users to get the a `PartialAppInfo` object with an application id. Uses the following path: `/applications/:id/rpc`

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/http.py [hunk 1]
```diff
@@ -1979,6 +1979,9 @@ def bulk_edit_guild_application_command_permissions(
 
     # Misc
 
+    def get_application(self, application_id: Snowflake, /) -> Response[appinfo.PartialAppInfo]:
+        return self.request(Route("GET", "/applications/{application_id}/rpc", application_id=application_id))
```
Comment (Rapptz) [inline discord/http.py:1983]:
```suggestion
        return self.request(Route('GET', '/applications/{application_id}/rpc', application_id=application_id))
```

Why is this using the /rpc endpoint? Where is this endpoint documented?

Comment (Soheab) [reply discord/http.py:1983]: Discord does not want to document it for some reason. According to [advaith in the discord-developers server](https://discord.com/channels/613425648685547541/697489244649816084/950540148033081354).

Comment (Rapptz) [reply discord/http.py]: I don't think I want to add an undocumented endpoint.

Comment (dolfies) [reply discord/http.py]: FWIW this endpoint has existed for many years and doesn't even need authentication.

Comment (Rapptz) [reply discord/http.py]: I'm sure it has, same with the activities endpoint and other undocumented ones. It's just these have no guarantee attached to them so if they break they stick with the library which has stronger versioning guarantees than Discord (e.g. look how long we had support for user accounts) and that's not good.

Comment (dolfies) [reply discord/http.py]: I agree that there's no guarantee. However, something like this is extremely useful and its better to have it with the small possibility it breaks than to not have it at all (IMO).

**Comment by Rapptz** [comment]
Closing since this is undocumented.

**Comment by lol219** [comment]
But the nsfw application command was undocumented too

**Comment by Soheab** [comment]
> But the nsfw application command was undocumented too and you added it.

That is an [upcoming feature](
https://gist.github.com/Stalruth/8258cd0cd7298ffaed6290bc7ecadb8e#apps-and-api-updates) for application commands. [This one won't ever be documented](https://github.com/discord/discord-api-docs/discussions/4620#discussioncomment-2385323) and is not something new.

**Comment by lol219** [comment]
> > But the nsfw application command was undocumented too and you added it.
> 
> That is an [upcoming feature](
> https://gist.github.com/Stalruth/8258cd0cd7298ffaed6290bc7ecadb8e#ending-questions-to-mason) for application commands. [This one won't ever be documented](https://github.com/discord/discord-api-docs/discussions/4620#discussioncomment-2385323) and is not something new.

Ok

**Pull request closed without merging**